src/Init/Data/List/BasicAux.lean

```diff
@@ -288,6 +306,15 @@ def mapMono (as : List α) (f : α → α) : List α :=
 Monadic generalization of `List.partition`.
 
 This uses `Array.toList` and which isn't imported by `Init.Data.List.Basic`.
+```
+def posOrNeg? (x : Int) : Option Bool :=
```

Comment:
Is there a better/canonical example which is used for monadic functions?

I also considered,
```lean
def posOrNeg (x : Int) : Except String Bool :=
  if x > 0 then pure true
  else if x < 0 then pure false
  else throw "Zero is not positive or negative
```

Reply:
You can use `IO` and `println!`

Reply:
What about this instead of the `Option Bool` example?
```lean
def isEven (x : Int) : IO Bool := do
  println! s!"Checking if {x} is even"
  pure (x % 2 == 0)

partitionM isEven [1, 2] = do
  println! "Checking if 1 is even"
  println! "Checking if 2 is even"
  pure ([2], [1])
```

Reply:
While this equality might hold, I think we'll have readers who aren't used to seeing things that way. I think the `Except` version is a better example, though the `IO` one could be an OK followup.

Reply:
Maybe it could be written not as an equality but just "this is printed when you `#eval` it"

Reply:
I like the `Except` version because `Except` is incorporated into the predicate logic. With two examples the docstring can illustrate how `partitionM` handles both successful partitions and exceptions:

```lean
partitionM posOrNeg [-1, 2, 3] = Except.ok ([2, 3], [-1])
partitionM posOrNeg [0, 2, 3] = Except.error "Zero is not positive or negative"
```

Additionally, here is an alternative `IO` example:

```lean
def isEven (x : Int) : IO Bool := do  
  println! s!"Checking if {x} is even"  
  pure (x % 2 == 0)

-- prints "Checking if 1 is even\nChecking if 2 is even\n"
#eval partitionM isEven [1, 2] => pure ([2], [1])
```

I would lean toward `Except` (or both if we want two examples).

Reply:
I think this `Except` example is great. No need to block this contribution on re-planning everything - let's look at the other functions on a case-by-case basis rather than deciding on an example monad _a priori_.

Reply:
Got it. I updated the proposed changes with the `Except` example.